src/librustdoc/config.rs

```diff
@@ -374,6 +377,31 @@ impl Options {
             }
         };
 
+        let mut default_settings: Vec<Vec<(String, String)>> = vec![
+            matches
+                .opt_str("default-theme")
+                .iter()
+                .map(|theme| {
+                    vec![
+                        ("use-system-theme".to_string(), "false".to_string()),
+                        ("theme".to_string(), theme.to_string()),
+                    ]
+                })
+                .flatten()
+                .collect(),
+            matches
+                .opt_strs("default-setting")
```

Comment:
One more thing that `--default-theme` breaks is you now have different behavior for `--default-theme x` and `--default-setting theme=x` - the first will set `use-system-theme=false`, the second will not.

Reply:
Yes.  That is inevitable and deliberate.  As you can see from the usage message we do not make any promises about the particular behaviour of `--default-setting`.  The available settings and their semantics and not documented and not stable.

Whereas the semantics of `--default-theme` are well-defined, and are implemented by setting the two internal settings.

I do think it is worth keeping `--default-setting` even despite these caveats because in practice the storage settings are visible to users via their browser and of course if they look at the JS.

The difference between `--default-setting` and `--default-theme` is like the difference between firefox's `about:config` and `about:preferences`.

Reply:
I guess my question is why you would ever want to modify `default-setting`. All those settings are also configurable on a per-user basis on [`settings.html`](https://doc.rust-lang.org/settings.html) - do we really need so much flexibility as to make the 'default defaults' configurable too?